spec/tapioca/dsl/compilers/protobuf_spec.rb

```diff
@@ -543,6 +543,119 @@ def fields=(value); end
               RBI
               assert_equal(expected, rbi_for("Google::Protobuf::Struct"))
             end
+
+            it "handles map and repeating types with anonymous subtype message classes by mapping them to T.untyped" do
+              add_ruby_file("protobuf.rb", <<~RUBY)
+                Google::Protobuf::DescriptorPool.generated_pool.build do
+                  add_file("cart.proto", :syntax => :proto3) do
+                    add_message "CartEntry" do
+                      optional :key, :string, 1
+                      optional :value, :string, 2
+                    end
+                    add_message "NonMapEntry" do
+                      optional :key, :string, 1
+                    end
+                    add_message "MyCart" do
+                      map :tables, :string, :message, 1, "CartEntry"
+                      repeated :items, :message, 2, "NonMapEntry"
+                    end
+                  end
+                end
+
+                Cart = Google::Protobuf::DescriptorPool.generated_pool.lookup("MyCart").msgclass
+                # we intentionally don't define `CartEntry` or `NonMapEntry` as constants, so that
+                # those message classes stay anonymous, which we expect to be mapped to `T.untyped`
+              RUBY
+
+              expected = <<~RBI
+                # typed: strong
+
+                class Cart
+                  sig { params(items: T.nilable(T.any(Google::Protobuf::RepeatedField[T.untyped], T::Array[T.untyped])), tables: T.nilable(T.any(Google::Protobuf::Map[String, T.untyped], T::Hash[String, T.untyped]))).void }
+                  def initialize(items: Google::Protobuf::RepeatedField.new(:message, T.untyped), tables: Google::Protobuf::Map.new(:string, :message, T.untyped)); end
```

Comment:
We should not be supplying these kinds of defaults, anyway. RBI does not care what the default value is, just that there is a default, so these should be `T.unsafe(nil)`.

I'll change that in the next PR.

Reply:
Btw, this does not "break" the RBI, since this file is not executable. If something were to run `Google::Protobuf::RepeatedField.new(:message, T.untyped)`, then it would fail, yes, but as it stands, there should be nothing wrong with it as a static construct.